src/kernels/installer/moduleInstaller.ts

```diff
@@ -55,25 +55,33 @@ export abstract class ModuleInstaller implements IModuleInstaller {
         const environmentService = this.serviceContainer.get<IEnvironmentVariablesService>(
             IEnvironmentVariablesService
         );
+        if (cancelTokenSource.token.isCancellationRequested) {
```

Comment:
You'll see these in a few places, while running the tests found that these were causing issues.
Basically we had stopped cell execution (either closing a document or interrupt), & these code paths would continue running.